### ISSUE TITLE
Issue with set_shape_transparency ifcopenshell

### ISSUE THREAD

**Issue by greeshmaunni1990** [IfcOpenShell/IfcOpenShell#1037]
Hi I was using the function ifcopenshell.geom.utils.set_shape_transparency(display, alpha) and was getting an error 'TypeError: AIS_InteractiveContext_SetTransparency expected 4 arguments, got 3'. So i have added the third agrument which is a boolean and set it to True. But now i am getting another error 'TypeError: in method 'AIS_InteractiveContext_SetTransparency', argument 2 of type 'opencascade::handle< AIS_InteractiveObject > const &''. Do you happen to know how to solve this issue??

**Comment by aothms** [comment]
H'm I indeed, can reproduce this on the latest 7.4 version of OCCT. Which is somewhat odd. One way around this is to not use AIS_InteractiveContext.SetTransparency() but instead AIS_Shape.SetTransparency() and potentially AIS_InteractiveContext.Update() afterwards.

What's the type of `ais` in your call to `ifcopenshell.geom.utils.set_shape_transparency()`. Can you add a `print(type(ais))` there?

Would be great if you can get back a working version and send a PR :)

**Comment by aothms** [comment]
Also: That first part of your comment should be handled by this( are you on an older version?) https://github.com/IfcOpenShell/IfcOpenShell/commit/e2e217700d68a89cde3e3a7c09ee13bd3c3ca66d

**Comment by greeshmaunni1990** [comment]
Hi Thank you for replying. The ooct i am using is of the version 7.4.0.and type of ais is "<class 'OCC.Display.OCCViewer.Viewer3d'>". I did not get portion where you were saying about changing to AIS_Shape.SetTransparency() as I am new to this. Could you explain?

**Comment by aothms** [comment]
> type of ais is "<class 'OCC.Display.OCCViewer.Viewer3d'>".

That's not the way it's intended to be used then. Call the function with a result obtained from `display_shape()` https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.6.0/src/ifcopenshell-python/ifcopenshell/geom/occ_utils.py#L106

The function currently uses `AIS_InteractiveContext.SetTransparency()` https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.6.0/src/ifcopenshell-python/ifcopenshell/geom/occ_utils.py#L198

But there is also a `AIS_InteractiveObject.SetTransparency()` https://old.opencascade.com/doc/occt-6.9.1/refman/html/class_a_i_s___interactive_object.html#a201080a4986f9403b97c87df92364177 Apperently that function is properly wrapped in PyOCC

**Comment by greeshmaunni1990** [comment]
Thank you fot you response. I have one more doubt i would like to ask you. I have some entities that has been shaded when user checks a checkbox. I have writen the code such that the shaded portion will be transp/arent when the user unchecks it. But the shadd portion is not getting transparent.I dont want to load the ifc code all over again. Is there any other method I could do it?

**Comment by arunkvijay** [comment]
> Also: That first part of your comment should be handled by this( are you on an older version?) [e2e2177](https://github.com/IfcOpenShell/IfcOpenShell/commit/e2e217700d68a89cde3e3a7c09ee13bd3c3ca66d)

This commit is missing in the latest ifcopenshell occ_utils.py and return the same error `TypeError: AIS_InteractiveContext_SetTransparency expected 4 arguments, got 3`

**Comment by aothms** [comment]
@arunkvijay You're right. It was mistakenly (?) removed again https://github.com/IfcOpenShell/IfcOpenShell/commit/e2784fb69b8e657eba3fe0625a4ce460cd0a60f6#diff-161ef287df58c3e2a27fe466533905a756cecfb9b8d44b5d9441d88229ba9b69L186-R194

I've added it back, but not with a lot of confidence. It's been three years and this code sees very little use apparently.
